ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -386,9 +437,15 @@ impl RawData for RawDataServerWrapper {
         );
 
         let output_stream = ReceiverStream::new(rx);
-        Ok(Response::new(
-            Box::pin(output_stream) as Self::GetTransactionsStream
-        ))
+        let mut response = Response::new(Box::pin(output_stream) as Self::GetTransactionsStream);
+
+        if let Some(connection_id) = request_metadata.request_connection_id.as_ref() {
```

Comment:
If request_connection_id doesn't exist in request_metadata, we should create and provide one and send it as part of the response.